torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -1037,17 +1048,25 @@ Tensor TensorExprKernel::bindInput(const torch::jit::Value* input) {
   auto const& outputs = input->owningGraph()->outputs();
   std::unordered_set<const Value*> outputs_set(outputs.begin(), outputs.end());
 
-  auto is_concrete_cont = [](const torch::jit::Value* input) {
+  auto is_concrete_cont = [](const torch::jit::Value* input,
+                             const MemoryLayoutPolicy& mem_layout_policy) {
     if (input->isCompleteTensor()) {
-      return isContiguous(input);
+      auto mem_layout = (mem_layout_policy == MemoryLayoutPolicy::kContiguous)
+          ? at::MemoryFormat::Contiguous
+          : at::MemoryFormat::ChannelsLast;
```

Comment:
I wonder if we can reuse `at::MemoryFormat` here so that we don't need to introduce `MemoryLayoutPolicy`.

Reply:
It looks like we don't really need to distinguish layouts based on the rank (if necessary, we can lookup the rank from the corresponding buffer, right?). Is there something I miss here?

Reply:
Suppose the graph contains 2 inputs while one is `at::MemoryFormat::ChannelsLast`(2d) and the other one is `at::MemoryFormat::ChannelsLast3d`, do you mean to use `at::MemoryFormat::ChannelsLast` to represent channels-last layout regardless the rank?